src/main/java/com/conveyal/datatools/manager/persistence/TypedPersistence.java

```diff
@@ -176,11 +188,20 @@ public List<T> getFiltered (Bson filter) {
      * Get all objects satisfying the supplied Mongo filter and sort by.
      */
     public List<T> getFiltered (Bson filter, Bson sortBy) {
-        if (sortBy != null) {
-            return mongoCollection.find(filter).sort(sortBy).into(new ArrayList<>());
-        } else {
-            return mongoCollection.find(filter).into(new ArrayList<>());
-        }
+        return (sortBy != null)
+        ? mongoCollection.find(filter).sort(sortBy).into(new ArrayList<>())
+        : mongoCollection.find(filter).into(new ArrayList<>());
+    }
+
+    /**
+     * Get all objects populating only the included fields satisfying the supplied Mongo filter and sort by.
+     */
+    public List<T> getFilteredLimitedFields(Bson filter, Bson sortBy, String... includedFields) {
```

Comment:
It's better than before but it's still quite slow... 

Is there any way to make the request only once on startup and then hold it? that should be less extreme than fully caching the results. we could fallback to making this request if we don't find the data in that initial request data.

I tried fiddling with mongo settings but couldn't get performance better enough